sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -310,6 +316,22 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog with QueryErrorsB
         ColumnDefinition.checkColumnDefinitions(operator)
 
         var stagedError: Option[() => Unit] = None
+
+        getAllExpressions(operator).foreach(_.foreach {
```

Comment:
there is another `getAllExpressions(operator).foreach(_.foreachUp {` below. Can we handle `ScopedExpression` there?

Reply:
moved into next block of checks

Reply:
Moving the checks into the next block makes the old error messages (wrong suggested columns) resurface:
- https://github.com/G-Research/spark/actions/runs/10684568288/job/29615369891#step:10:31013
- https://github.com/G-Research/spark/actions/runs/10684568288/job/29615369891#step:10:31076
- https://github.com/G-Research/spark/actions/runs/10684568288/job/29615369891#step:10:31139
- https://github.com/G-Research/spark/actions/runs/10684568288/job/29615369891#step:10:31202

Even if I move the checks to the top of the block. So they have to be checked in isolation of the other checks.

Reply:
reverted 66d3471efea7dad49e915054d7480521689d9b3f